app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,32 @@
+class OrdersController < ApplicationController
+  def show
+    if !current_user || current_user != (@order = Order.find(params[:id])).user
```

Comment:
Use 
```ruby
before_action :require_sign_in!, only: [:show, :create]
before_action :set_order, only: [:show]`
```
It's kind of standard to do that.

In `require_sign_in`, you probably want to redirect user to the sign-in page with a flash error message. This `@access_denied = true` is non-standard.